deploy/terraform/main.tf

```diff
@@ -10,43 +10,46 @@ provider "packet" {
 }
 
 # Create a new VLAN in datacenter "ewr1"
-resource "packet_vlan" "provisioning-vlan" {
-  description = "provisioning-vlan"
+resource "packet_vlan" "provisioning_vlan" {
+  description = "provisioning_vlan"
   facility    = var.facility
   project_id  = var.project_id
 }
 
 # Create a device and add it to tf_project_1
-resource "packet_device" "tink-provisioner" {
+resource "packet_device" "tink_provisioner" {
   hostname         = "tink-provisioner"
   plan             = var.device_type
   facilities       = [var.facility]
   operating_system = "ubuntu_18_04"
   billing_cycle    = "hourly"
   project_id       = var.project_id
   user_data        = file("install_package.sh")
+}
+
+resource "null_resource" "tink_directory" {
+  connection {
+    type = "ssh"
+    user = var.ssh_user
+    host = packet_device.tink_provisioner.network[0].address
+  }
 
   provisioner "file" {
     source      = "./../../../tink"
     destination = "/root/"
-
-    connection {
-      type        = "ssh"
-      user        = var.ssh_user
-      host        = packet_device.tink-provisioner.network[0].address
-      private_key = file(var.ssh_private_key)
```

Comment:
@gianarb I removed the `private_key` because Terraform will use your ssh-agent by default.

The problem with defaulting to "~/.ssh/id_rsa" is that this file is usually encrypted, and Terraform can not use that.

My suggestion is that we ask the user to add the key they want to use to their SSH agent before running `terraform apply`.  For most users, this should already be the case.

An alternative is to ask users to `ssh-keygen` create a new key in this directory, and tell Terraform to use that.

If we can make `private_key` optional (use the agent if ssh_private_key is empty) that would be even better.  I can look into this.

Reply:
I added the `ssh-agent` note as part of the documentation. We can do the fallback work as follow up work 👍